### PR TITLE
Use one constructor for NativeReanimatedModule with bridgeless

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

@tomekzaw wanted this badly

### CODE REVIEW

File: Common/cpp/NativeModules/NativeReanimatedModule.cpp [hunk 1]
```diff
@@ -44,65 +44,38 @@ bool CoreFeatures::useNativeState;
...
+bool JSSchedulerDependencyHolder::isBridgeless() const {
+  return runtimeExecutor_ != nullptr;
+}
+
+const std::shared_ptr<CallInvoker> JSSchedulerDependencyHolder::getJsInvoker() const {
+  return jsInvoker_;
+}
+
+RuntimeExecutor JSSchedulerDependencyHolder::getRuntimeExecutor() const {
+  return runtimeExecutor_;
```
Comment (tomekzaw) [inline Common/cpp/NativeModules/NativeReanimatedModule.cpp:59]:
```suggestion
  react_native_assert(runtimeExecutor_ != nullptr);
  return runtimeExecutor_;
```
or
```suggestion
  react_native_assert(isBridgeless());
  return runtimeExecutor_;
```

File: android/src/main/cpp/NativeProxy.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ NativeProxy::NativeProxy(
       rnRuntime_(rnRuntime),
       nativeReanimatedModule_(std::make_shared<NativeReanimatedModule>(
           *rnRuntime,
-          jsCallInvoker,
+          JSSchedulerDependencyHolder(jsCallInvoker),
```
Comment (tomekzaw) [inline android/src/main/cpp/NativeProxy.cpp:47]:
```suggestion
          std::make_shared<JSScheduler>(rnRuntime, jsCallInvoker),
```

File: apple/native/NativeProxy.mm [hunk 3]
```diff
@@ -64,8 +64,10 @@ - (void *)runtime;
 
   PlatformDepMethodsHolder platformDepMethodsHolder = makePlatformDepMethodsHolder(bridge, nodesManager, reaModule);
 
+  JSSchedulerDependencyHolder jsSchedulerDependencyHolder(jsInvoker);
```
Comment (tomekzaw) [inline apple/native/NativeProxy.mm:67]:
```suggestion
  auto jsScheduler = std::make_shared<JSScheduler>(rnRuntime, jsInvoker);
```

File: apple/native/NativeProxy.mm [hunk 4]
```diff
@@ -64,8 +64,10 @@ - (void *)runtime;
...
+
   auto nativeReanimatedModule = std::make_shared<NativeReanimatedModule>(
-      rnRuntime, jsInvoker, jsQueue, uiScheduler, platformDepMethodsHolder, valueUnpackerCode);
+      rnRuntime, jsSchedulerDependencyHolder, jsQueue, uiScheduler, platformDepMethodsHolder, valueUnpackerCode);
```
Comment (tomekzaw) [inline apple/native/NativeProxy.mm:70]:
```suggestion
      rnRuntime, jsScheduler, jsQueue, uiScheduler, platformDepMethodsHolder, valueUnpackerCode);
```

**Review by tomekzaw — CHANGES_REQUESTED** [review]
@tjzel Thanks again for your effort.

General feedback: Why don't we just instantiate `JSScheduler` before calling `NativeReanimatedModule` ctor? This way we can avoid adding `JSScheduleDependencyHolder` to our codebase class at all.

The only obstacle is that we need to pass `rnRuntime` along with `jsInvoker` (in bridgeful mode) but this is not a real problem as we luckily have access to it without any problem. In bridgeless mode, we don't need `rnRuntime` at all.

File: Common/cpp/Tools/JSScheduler.h [hunk 5]
```diff
@@ -37,6 +37,7 @@ class JSScheduler {
 
  public:
   const std::function<void(Job)> scheduleOnJS = nullptr;
```
Comment (tomekzaw) [inline Common/cpp/Tools/JSScheduler.h:39]: Can we make it private?

File: Common/cpp/Tools/JSScheduler.h [hunk 6]
```diff
@@ -37,6 +37,7 @@ class JSScheduler {
...
+  const std::shared_ptr<CallInvoker> getJSCallInvoker() const;
```
Comment (tomekzaw) [inline Common/cpp/Tools/JSScheduler.h:40]: Let's move this declaration right after `JSScheduler` constructor (we already have `public:` there).

File: android/src/main/cpp/NativeProxy.cpp [hunk 7]
```diff
@@ -44,11 +44,12 @@ NativeProxy::NativeProxy(
       rnRuntime_(rnRuntime),
       nativeReanimatedModule_(std::make_shared<NativeReanimatedModule>(
           *rnRuntime,
-          jsCallInvoker,
+          std::make_shared<JSScheduler>(*rnRuntime, jsCallInvoker),
```
Comment (tomekzaw) [inline android/src/main/cpp/NativeProxy.cpp:47]: Let's construct `JSScheduler` before:

```cpp
auto jsScheduler = std::make_shared<JSScheduler>(*rnRuntime, jsCallInvoker);
```

File: android/src/main/cpp/NativeProxy.cpp [hunk 8]
```diff
@@ -65,17 +66,17 @@ NativeProxy::NativeProxy(
...
     jni::alias_ref<facebook::react::JFabricUIManager::javaobject>
         fabricUIManager,
-
     const std::string &valueUnpackerCode)
     : javaPart_(jni::make_global(jThis)),
       rnRuntime_(rnRuntime),
       nativeReanimatedModule_(std::make_shared<NativeReanimatedModule>(
           *rnRuntime,
-          runtimeExecutor,
+          std::make_shared<JSScheduler>(*rnRuntime, runtimeExecutor),
```
Comment (tomekzaw) [inline android/src/main/cpp/NativeProxy.cpp:74]: same as above

File: android/src/main/cpp/NativeProxy.cpp (hunk 7 above)
Comment (tjzel) [reply android/src/main/cpp/NativeProxy.cpp:47]: no can do (it's an initializer list)

File: android/src/main/cpp/NativeProxy.cpp (hunk 8 above)
Comment (tjzel) [reply android/src/main/cpp/NativeProxy.cpp:74]: no can do (it's an initializer list)

File: android/src/main/cpp/NativeProxy.cpp [hunk 9]
```diff
@@ -65,17 +66,17 @@ NativeProxy::NativeProxy(
...
           std::make_shared<JMessageQueueThread>(messageQueueThread),
           uiScheduler,
           getPlatformDependentMethods(),
-          valueUnpackerCode)),
+          valueUnpackerCode,
+          true)),
```
[the remark this answers is not in the document]
Comment (tjzel) [reply android/src/main/cpp/NativeProxy.cpp:79]: no can do (it's an initializer list)

File: Common/cpp/Tools/JSScheduler.cpp [hunk 10]
```diff
@@ -27,4 +28,10 @@ JSScheduler::JSScheduler(
       }) {}
 #endif // REACT_NATIVE_MINOR_VERSION >= 74 && defined(RCT_NEW_ARCH_ENABLED
 
+// DO NOT USE OR YOU WILL BE PROMOTED
```
[the remark this answers is not in the document]
Comment (tjzel) [reply Common/cpp/Tools/JSScheduler.cpp]: 😢

**Review by tomekzaw — APPROVED** [review]
Thanks for fixing this! ❤️

**Pull request merged**